cron/monthly/user-report.js

```diff
@@ -198,25 +199,25 @@ const processBacker = async FromCollectiveId => {
         createdAt: { [Op.gte]: startDate, [Op.lt]: endDate },
         SubscriptionId: { [Op.ne]: null },
       },
+      deletedAt: null,
     },
     include: [{ model: models.Subscription }],
   });
-  const ordersByCollectiveId = {};
-  orders.map(o => {
-    ordersByCollectiveId[o.CollectiveId] = o;
-  });
+  // group orders(by collective) that either don't have subscription or have active subscription
+  const ordersByCollectiveId = groupBy(orders.filter(o => !o.Subscription || o.Subscription.isActive), 'CollectiveId');
```

Comment:
It might be possible that we have collectives with multiple donations each month (multiple single time donation,  mix of subscription + single time, and even multiple subscriptions).

How do we handle that?

Reply:
This script was considering only subscription orders. At first I thought it was intentional but It makes senses to cover all orders regardless. I'll update the code to reflect it.

Reply:
Sorry, i explained wrong. I meant it was only getting the first order of each collective which would ONLY make sense if it was considering subscriptions only(afaik a contributor is supposed to have only one subscription per collective). The following part of the master user-report is wrong because it's basically iterating over the `Order.findAll` query and if there is more than one order of the same collective it would consider only the latest one(also there was not filter to check whether the subscription was active anymore) :
https://github.com/opencollective/opencollective-api/blob/master/cron/monthly/user-report.js#L204-L207